docs/installation.rst

```diff
@@ -67,11 +67,23 @@ Installation with NVIDIA GPU support (CUDA)
 
 .. note::
 
-   To use JAX on the GPU, make sure that your ``jaxlib`` is `built with CUDA support <https://github.com/google/jax#installation>`_.
+   There are 2 ways to install jax with CUDA support:
+   - using a pypi-distributed CUDA installation (suggested by jax developers) ``pip install -U 'jax[cuda12_pip]' -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html`` 
```

Comment:
Not `jax[cuda]`?

Reply:
`jax[cuda]` does not exist.
`jax[cuda12]` does exist, but won't work because of other bugs in mpi4jax / recent changes in XLA primitive registration (or at least, I could not get it to work).

If we fix those bugs I will update the guide.

Reply:
Should we add a warning then not to use `jax[cuda12]`? This is the recommended mode of installing JAX after all (while `jax[cuda12_pip]` is deprecated per the docs).

Reply:
added relevant notices